libc/string_mem.c

```diff
@@ -0,0 +1,404 @@
+/*
+ * Phoenix-RTOS
+ *
+ *    libc-tests
+ *    HEADER:
+ *    - string.h
+ *    TESTED:
+ *    - memset(),
+ *    - memmove()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Adam Debek
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <string.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <ulimit.h>
+#include <stdint.h>
+#include <unity_fixture.h>
+
+#define BUF_LEN         16
+#define VALUES_NUM      16
+#define MEMS_NUM        8
+#define BIG_SIZE        1024
+#define TEST_STR        "THIS IS TEST MEM123"
+#define TEST_STR_SIZE   20
+#define OFFSET          40
+#define OFFSET_BIG      1500
+#define MEM_SIZE        100
+#define MEM_SIZE_BIG    4000
+#define OVERLAP_LEN     (TEST_STR_SIZE / 2)
+#define OVERLAP_LEN_BIG (BIG_SIZE / 2)
+
+static int8_t buf[BUF_LEN];
+static int8_t buf_byte[256];
+static int8_t buf_byte_exp[256];
+static void *ret;
+static void *mem;
+static void *mem_big;
+static int8_t big_buf[1024];
+
+static int8_t expected_mem[MEMS_NUM][BUF_LEN] = {
+	{ -127, -127, -127, -127, -127, -127, -127, -127, -127, -127, -127, -127, -127, -127, -127, -127 },
+	{ -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1 },
+	{ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 },
+	{ 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1 },
+	{ 127, 127, 127, 127, 127, 127, 127, 127, 127, 127, 127, 127, 127, 127, 127, 127 },
+	{ 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128 },
+	{ 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200, 200 },
+	{ 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255 }
+};
+
+static int8_t expect_mem_big[BIG_SIZE];
+
+static int values[VALUES_NUM] = { -127, -1, 0, 1, 127, 128, 200, 255,
+	385, 511, 768, 257, 383, 384, 456, 511 };
+/* next values exceed 1 byte, (int8_t)385 = -127, (int8_t)511 = -1 and so on */
+
+
+typedef struct over_write {
+	int32_t val1;
+	int32_t val2;
+} over_write_t;
+
+
+TEST_GROUP(string_memset);
```

Comment:
The new approach. Thanks to it we can see all test groups provided in this file.

```suggestion
TEST_GROUP(string_memset);
TEST_GROUP(string_memmove);
TEST_GROUP(string_memmove_big);
```